src/solver/conv_hip_implicit_gemm_bwd_v1r1_xdlops_nchw_kcyx_nkhw.cpp

```diff
@@ -0,0 +1,978 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/conv/invokers/impl_gemm.hpp>
+#include <miopen/solver.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/generic_search.hpp>
+
+#include "implicitgemm_util.hpp"
+
+#include <cstddef>
+
+namespace miopen {
+namespace solver {
+
+PerformanceImplicitGemmBwdV1R1Xdlops::PerformanceImplicitGemmBwdV1R1Xdlops()
+    : PerformanceImplicitGemmBwdV1R1Xdlops::PerformanceImplicitGemmBwdV1R1Xdlops(
+          4, 4, 1, 4, 4, 1, false, false)
+{
+}
+
+PerformanceImplicitGemmBwdV1R1Xdlops::PerformanceImplicitGemmBwdV1R1Xdlops(
+    int GemmMPerBlock_,
+    int GemmNPerBlock_,
+    int GemmKPerBlock_,
+    int GemmMPerWave_,
+    int GemmNPerWave_,
+    int GemmKPack_,
+    bool GemmAThreadCopyMoreGemmK_,
+    bool GemmBThreadCopyMoreGemmKPack_)
+    : GemmMPerBlock(GemmMPerBlock_),
+      GemmNPerBlock(GemmNPerBlock_),
+      GemmKPerBlock(GemmKPerBlock_),
+      GemmMPerWave(GemmMPerWave_),
+      GemmNPerWave(GemmNPerWave_),
+      GemmKPack(GemmKPack_),
+      GemmAThreadCopyMoreGemmK(GemmAThreadCopyMoreGemmK_),
+      GemmBThreadCopyMoreGemmKPack(GemmBThreadCopyMoreGemmKPack_)
+{
+}
+
+bool PerformanceImplicitGemmBwdV1R1Xdlops::
+operator==(const PerformanceImplicitGemmBwdV1R1Xdlops& other) const
+{
+    // clang-format off
+    return GemmMPerBlock == other.GemmMPerBlock
+        && GemmNPerBlock == other.GemmNPerBlock
+        && GemmKPerBlock == other.GemmKPerBlock
+        && GemmMPerWave == other.GemmMPerWave
+        && GemmNPerWave == other.GemmNPerWave
+        && GemmKPack == other.GemmKPack 
+        && GemmAThreadCopyMoreGemmK  == other.GemmAThreadCopyMoreGemmK
+        && GemmBThreadCopyMoreGemmKPack  == other.GemmBThreadCopyMoreGemmKPack;
+    // clang-format on
+}
+
+bool PerformanceImplicitGemmBwdV1R1Xdlops::SetNextValue()
+{
+    do
+    {
+        // list performance parameters in reverse order, in order for tuning to iterate over the
+        // range in normal order
+        if(!NextFlag<false, true>(GemmBThreadCopyMoreGemmKPack))
+            break;
+        if(!NextFlag<false, false>(GemmAThreadCopyMoreGemmK))
```

Comment:
```GemmAThreadCopyMoreGemmK``` is always false during tuning in fwd-v4r4, because I found ```false``` to be always faster.

I'm not sure it's the same case for bwd-data-v1r1

Reply:
Is there an conclusion whether it's worth to make it tunable?

Reply:
Inceptionv3 data says it is worth tunable. I made it tunable. 

1|ConvHipImplicitGemmBwdDataV1R1Xdlops|1|gfx908|120|128,128,2,64,64,8,1,1
2|ConvHipImplicitGemmBwdDataV1R1Xdlops|2|gfx908|120|128,128,2,64,64,4,0,1
3|ConvHipImplicitGemmBwdDataV1R1Xdlops|3|gfx908|120|128,128,8,64,64,8,1,1
4|ConvHipImplicitGemmBwdDataV1R1Xdlops|4|gfx908|120|128,128,2,64,64,4,0,1
5|ConvHipImplicitGemmBwdDataV1R1Xdlops|5|gfx908|120|128,128,4,64,64,4,1,0
6|ConvHipImplicitGemmBwdDataV1R1Xdlops|6|gfx908|120|128,128,4,64,64,4,1,0
7|ConvHipImplicitGemmBwdDataV1R1Xdlops|7|gfx908|120|128,128,4,64,64,4,0,1
8|ConvHipImplicitGemmBwdDataV1R1Xdlops|8|gfx908|120|128,128,4,64,64,4,1,0
9|ConvHipImplicitGemmBwdDataV1R1Xdlops|9|gfx908|120|32,128,4,32,64,4,1,0
10|ConvHipImplicitGemmBwdDataV1R1Xdlops|10|gfx908|120|32,128,1,32,64,8,0,0
11|ConvHipImplicitGemmBwdDataV1R1Xdlops|11|gfx908|120|64,128,2,32,64,8,1,1
12|ConvHipImplicitGemmBwdDataV1R1Xdlops|12|gfx908|120|64,128,4,32,64,8,0,0
13|ConvHipImplicitGemmBwdDataV1R1Xdlops|13|gfx908|120|64,128,4,32,64,4,1,1
14|ConvHipImplicitGemmBwdDataV1R1Xdlops|14|gfx908|120|64,128,4,32,64,8,0,1
15|ConvHipImplicitGemmBwdDataV1R1Xdlops|15|gfx908|120|64,128,4,64,32,8,1,1
16|ConvHipImplicitGemmBwdDataV1R1Xdlops|16|gfx908|120|64,128,2,64,32,8,1,1
17|ConvHipImplicitGemmBwdDataV1R1Xdlops|17|gfx908|120|64,128,4,32,64,4,1,1
18|ConvHipImplicitGemmBwdDataV1R1Xdlops|18|gfx908|120|128,128,8,64,64,8,0,1
19|ConvHipImplicitGemmBwdDataV1R1Xdlops|19|gfx908|120|128,128,8,64,64,4,0,0
20|ConvHipImplicitGemmBwdDataV1R1Xdlops|20|gfx908|120|128,128,4,64,64,8,1,0
21|ConvHipImplicitGemmBwdDataV1R1Xdlops|21|gfx908|120|128,128,4,64,64,8,0,0
22|ConvHipImplicitGemmBwdDataV1R1Xdlops|22|gfx908|120|128,128,2,64,64,4,1,1
23|ConvHipImplicitGemmBwdDataV1R1Xdlops|23|gfx908|120|128,128,2,64,64,4,1,1
24|ConvHipImplicitGemmBwdDataV1R1Xdlops|24|gfx908|120|32,128,2,32,64,8,1,0
25|ConvHipImplicitGemmBwdDataV1R1Xdlops|25|gfx908|120|32,128,2,32,64,4,0,1
26|ConvHipImplicitGemmBwdDataV1R1Xdlops|26|gfx908|120|128,128,8,64,64,4,1,1
27|ConvHipImplicitGemmBwdDataV1R1Xdlops|27|gfx908|120|128,128,1,64,64,4,0,1
28|ConvHipImplicitGemmBwdDataV1R1Xdlops|28|gfx908|120|64,128,4,64,32,8,1,0
29|ConvHipImplicitGemmBwdDataV1R1Xdlops|29|gfx908|120|64,128,4,32,64,8,1,1
30|ConvHipImplicitGemmBwdDataV1R1Xdlops|30|gfx908|120|64,128,2,64,32,8,1,1
31|ConvHipImplicitGemmBwdDataV1R1Xdlops|31|gfx908|120|128,128,4,64,64,4,1,1
32|ConvHipImplicitGemmBwdDataV1R1Xdlops|32|gfx908|120|128,128,2,64,64,8,1,1
33|ConvHipImplicitGemmBwdDataV1R1Xdlops|33|gfx908|120|128,128,8,64,64,8,1,1
34|ConvHipImplicitGemmBwdDataV1R1Xdlops|34|gfx908|120|32,128,8,32,32,4,1,1
35|ConvHipImplicitGemmBwdDataV1R1Xdlops|35|gfx908|120|32,128,4,32,64,4,1,0